mirrormanager2/app.py

```diff
@@ -1361,8 +1364,74 @@ def shutdown_session(exception=None):
     """ Remove the DB session at the end of each request. """
     SESSION.remove()
 
+
 # pylint: disable=W0613
 @APP.before_request
 def set_session():
     """ Set the flask session as permanent. """
     flask.session.permanent = True
+
+
+def statistics_file_name(date, cat, ext):
+    year = date.year
+    month = date.month
+    day = date.day
+    name = "%s/%s/%s-" % (year, month, cat)
+    name = "%s%s-%s-%s.%s" % (name, year, month, day, ext)
+    return name
+
+
+def check_for_statistics(date, cat):
+    try:
+        stat_file = APP.config['STATISTICS_BASE']
+        stat_file = os.path.join(
+            stat_file,
+            statistics_file_name(date, cat, 'txt')
+        )
+        os.stat(stat_file)
```

Comment:
Yes, this is a bit strange. The function assumes some things about its environment, which could be made clearer. The function basically wants to know if the file exists or not. If the stat() call fails an exception is thrown and the calling function gets a None back.

The function assumes about its environment that it is set up that if the stat() call works the file actually can also be read (which must not be true, but it is assumed). So, yes this could be better. I will change it to actually check if the file actually can be read.